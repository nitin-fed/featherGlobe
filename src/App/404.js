import React, { Component } from 'react';

const MissingPage = () => {
    return(
        <div> Missing Page</div>
    )
}

export default MissingPage;