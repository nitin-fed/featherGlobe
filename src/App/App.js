import React, { Component } from "react";
import Navigation from "./Navigation/navigation";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import {store} from './Store/createStore';
import Logo from "./Components/Logo";
import "bootstrap/dist/css/bootstrap.min.css";

class App extends Component {
   render() {
    return (
      <Provider store={store}>
        <BrowserRouter>
          <div className='mainWrapper'>
              <Logo />
              <Navigation />
            <footer className="footer">www.featherglobe.com</footer>
          </div>
        </BrowserRouter>
      </Provider>
    );
  }
}

export default App;