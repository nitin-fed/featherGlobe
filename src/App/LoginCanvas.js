

export const init = () => {
  const canvas = document.querySelector("#myCanvas");

  const canvasWidth = 500;
  const canvasHeight = 300;
  const canvasBackground = "#e99e9e";
  const circleCount = 50;

  canvas.width = canvasWidth;
  canvas.height = canvasHeight;
  canvas.style.background = canvasBackground;

  window.addEventListener("resize", () => {
    canvas.width = canvasWidth;
    canvas.height = canvasHeight;
  });

  const drawCircle = () => {
    var ctx = canvas.getContext("2d");
    var rect = canvas.getBoundingClientRect();

    const circleArr = [];
    for (let i = 0; i < circleCount; i++) {
      var x = Math.random() * canvasWidth;
      var y = Math.random() * canvasHeight;
      var dx = Math.random() * -0.5 * 4;
      var dy = Math.random() * -0.5 * 8;
      const radius = Math.random() * 10;
      const randomColor = "hsl(" + 360 * Math.random() + ",100%,50%)";
      const circle = new Circle(x, y, dx, dy, radius, randomColor);
      circleArr.push(circle);
    }

    const animate = () => {
      requestAnimationFrame(animate);
      ctx.clearRect(0, 0, canvasWidth, canvasHeight);
      for (let i = 0; i < circleArr.length; i++) {
        circleArr[i].update();
      }
      ctx.save();
    };
    animate();

    function Circle(x, y, dx, dy, radius, cirColor) {
      this.x = x;
      this.y = y;
      this.dx = dx;
      this.dy = dy;
      this.radius = radius;

      this.draw = function() {
        ctx.beginPath();
        ctx.arc(this.x, this.y, this.radius, 0, 2 * Math.PI);
        ctx.fillStyle = cirColor;
        ctx.lineWidth = 1;
        // ctx.strokeStyle = "#FF0000";
        // ctx.stroke();
        ctx.fill();
      };

      this.update = function() {
        if (this.x + this.radius > canvasWidth || this.x - this.radius < 0) {
          this.dx = -this.dx;
        }

        if (this.y + this.radius > canvasHeight || this.y - this.radius < 0) {
          this.dy = -this.dy;
        }
        this.x += this.dx;
        this.y += this.dy;

        this.draw();
      };
    }
  };

  canvas.addEventListener("click", () => {
    drawCircle();
  });
};
