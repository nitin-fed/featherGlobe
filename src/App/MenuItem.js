

import React from "react";

import { primaryButtonStyle } from "./Utils/constants";

export default function MenuItem({ item , handleMenuItemClick}) {
  console.log(item);
  return (
    <li>
      <button onClick={handleMenuItemClick} className={primaryButtonStyle}> {item}</button>
    </li>
  );
}
