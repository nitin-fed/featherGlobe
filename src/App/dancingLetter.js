

export const initDancingLetter = () => {
  const canvas = document.querySelector("#dancingLetter");

  const canvasWidth = 500;
  const canvasHeight = 300;
  const canvasBackground = "#e7e7e7";

  canvas.width = canvasWidth;
  canvas.height = canvasHeight;
  canvas.style.background = canvasBackground;

  window.addEventListener("resize", () => {
    canvas.width = canvasWidth;
    canvas.height = canvasHeight;
  });

  const drawLetter = () => {
    var ctx = canvas.getContext("2d");

    ctx.font = "60px Arial";
    ctx.lineWidth = 2;
    ctx.strokeStyle = "#FF0000";
    ctx.translate(20, 50); // set canvas context to center
    ctx.rotate((Math.PI / 180) * 10); // 1/2 a degree
    //ctx.rotate((3 * Math.PI) / 180);
    ctx.fillText("A", 20, 50);
  };

  canvas.addEventListener("click", () => {
    drawLetter();
  });
};
