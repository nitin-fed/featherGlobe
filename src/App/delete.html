<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>

</body>

<script>
    function getFriday(day, month, year)
    {

        date = new Date(year, month - 1, day)

        if (date.getDay() === 5)
        {
            return Math.floor(((day - 1) / 7) + 1)
        } else
        {
            return "Date is not a Friday"
        }
    }

    console.log(getFriday(6, 8, 2021))



    // const givenDate = 'Mar 22, 2021'
    // const dateObj = new Date(givenDate)
    // const date = dateObj.getDate()
    // const dayOfFirst = new Date(dateObj.getMonth() + 1 + "," + 01 + "," + dateObj.getFullYear()).getDay()
    // const dateOfFirstFri = 5 - (dayOfFirst - 1)
    // if (dateOfFirstFri == date)
    // {
    //     console.log("1st Friday")
    // } else
    // {
    //     for (let i = 1; i < 5; i++)
    //     {
    //         if (dateOfFirstFri + (7 * i) == date)
    //         {
    //             console.log(i + 1);
    //         }
    //     }
    // }

</script>

</html>