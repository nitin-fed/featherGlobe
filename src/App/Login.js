/** @format */

import React, { useEffect, useRef, useState } from "react";
import { NavLink, useNavigate, useLocation } from "react-router-dom";
import { auth } from "../firebase-config";
import { TextInputField } from "./Components/TextInputField";
import {
  primaryButtonStyle,
  warningButtonStyle,
  secondaryButtonStyle
} from "./Utils/constants";
import {
  createUserWithEmailAndPassword,
  signOut,
  signInWithEmailAndPassword
} from "firebase/auth";
import { useDispatch, useSelector } from "react-redux";
import { updateCurrentUser } from "./Store/Reducers/appSlice";
import { init } from "./LoginCanvas";
import { initDancingLetter } from "./dancingLetter";

const Login = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const history = useNavigate();
  const dispatch = useDispatch();

  const [isLogin, setIsLogin] = useState();
  const [error, setError] = useState(null);
  const [isLoading, setLoading] = useState(false);

  useEffect(() => {
    console.log(location.pathname);
    if (location.pathname === "/registerUser") {
      setIsLogin(false);
    } else {
      setIsLogin(true);
    }
  }, [location.pathname]);

  // const login = () => {
  //   localStorage.setItem("user", "test");
  //   navigate("/dashboard");
  // };

  const userNameRef = useRef(null);
  const passwordRef = useRef(null);

  const allRefs = [{ password: passwordRef }, { userName: userNameRef }];

  const handleSubmit = async (evt) => {
    evt.preventDefault();
    const email = userNameRef.current.getValue();
    const password = passwordRef.current.getValue();
    setLoading(true);
    try {
      debugger;
      const user = await signInWithEmailAndPassword(
        auth,
        email.toString(),
        password
      );
      history("/");
    } catch (error) {
      setError(error.message);
    }
    setLoading(false);
    // evt.preventDefault();
    // const email = userNameRef.current.getValue();
    // const password = passwordRef.current.getValue();
    // console.log(isLoading);
    // try {
    //   setLoading(true);
    //   const user = await createUserWithEmailAndPassword(auth, email, password);
    //   dispatch(updateCurrentUser(user.user.email));
    // } catch (error) {
    //   setLoading(false);
    //   setError(error.message);
    // }
  };

  const handleLogin = async (evt) => {
    evt.preventDefault();
    const email = userNameRef.current.getValue();
    const password = passwordRef.current.getValue();
    setLoading(true);
    try {
      const user = await signInWithEmailAndPassword(auth, email, password);
      history("/");
    } catch (error) {
      setError(error.message);
    }
    setLoading(false);
  };

  useEffect(() => {
    auth.onAuthStateChanged((user) => {
      dispatch(updateCurrentUser(user?.email));
    });
  }, []);

  const handleReset = (e) => {
    e && e.preventDefault();
    setError();

    allRefs.forEach((item) => {
      Object.values(item).forEach((value) => {
        value.current && value.current.resetField();
      });
    });
  };

  const handleCancel = () => {
    history("/");
  };

  useEffect(() => {
    // init();
    //initDancingLetter();
  }, []);

  return (
    <div className='login p-10 sm:w-full lg:w-3/4 m-auto'>
      <h1 className='text-3xl'>{isLogin ? " Login" : "Register User"}</h1>
      <div className='text-red-500 py-2'>{error && error}</div>
      <form onSubmit={(evt) => handleSubmit(evt)}>
        <TextInputField
          label='User Name'
          kind='text'
          ref={userNameRef}
          value=''
          isRequired={true}
          validation={["required", "min", "max", "email"]}
        />

        <TextInputField
          label='Password'
          kind='password'
          ref={passwordRef}
          value=''
          isRequired={true}
          validation={["required", "min", "max"]}
        />

        {!isLogin && (
          <TextInputField
            label='Confirm Password'
            kind='confirmPassword'
            ref={passwordRef}
            value=''
            isRequired={true}
            validation={["required", "min", "max"]}
          />
        )}

        <br />
        <div className='float-right'>
          <button
            onClick={(e) => handleCancel(e)}
            className={secondaryButtonStyle}
          >
            Cancel
          </button>
          <button
            onClick={(e) => handleReset(e)}
            className={secondaryButtonStyle}
          >
            Reset
          </button>
          {isLogin ? (
            <button disabled={isLoading} className={primaryButtonStyle}>
              Login
            </button>
          ) : (
            <button
              disabled={isLoading}
              type={"submit"}
              className={primaryButtonStyle}
            >
              Sign up
            </button>
          )}
        </div>
      </form>
      <br />
      <br />
      {isLogin && (
        <div>
          Don't have account?
          <NavLink to='/registerUser' className='links px-2 text-sm'>
            Register now using email
          </NavLink>
          |
          <NavLink to='#' className='links px-2 text-sm'>
            Forgot Password
          </NavLink>
        </div>
      )}
      {/* <canvas id='myCanvas'></canvas> <canvas id='dancingLetter'></canvas> */}
    </div>
  );
};

export default Login;
