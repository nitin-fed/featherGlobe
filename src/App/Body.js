/** @format */

import React, { useEffect } from "react";

import { Route, Routes } from "react-router-dom";

// import Backdrop from "../Components/Backdrop";
import GuessedWordWrapper from "./Demo/Container/GuessedWordWrapper";
import { JottoApp } from "./Jotto/JottoApp";
import Article from "./Demo/Article/Article";
import Gallery from "./Demo/Container/Gallery";
import Users from "./Demo/Components/Users/Users";
import Support from "./Demo/Support/Support";
import Photos from "./Photography/Components/Photos";
import Posts from "./Demo/Container/Posts";
import Counters from "./Demo/Container/counters";
import { About } from "../App/About/about";
import { Contact } from "../App/About/Components/contact";

// import RegisterUser from "./Demo/Container/Signup";
// import Contact from "./Profile/Components/contact";
import Galaxy from "./Demo/Components/galaxy";
import Protected from "../Utils/Protected";
import PublicRoutes from "../Utils/PublicRoutes";
import Login from "./Login";
import { Profile } from "./Demo/Container/Profile";
import { UnAuthUser } from "./Components/UnAuthUser";

import { useDispatch } from "react-redux";
import { updateCurrentUser } from "./Store/Reducers/appSlice";
import { auth } from "../firebase-config";
import { Post } from "./Demo/Components/Post";
import { Admin } from "./Components/Admin";

import SecHome from "../sec";
import { UploadPhotos } from "./About/Components/uploadPhotos";

export const Body = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    auth.onAuthStateChanged((user) => {
      dispatch(updateCurrentUser(user?.email));
    });
  }, []);

  return (
    <Routes>
      {/* <Route exact path='/' element={<Home />} /> */}
      <Route path='/photography' element={<Photos />} />
      <Route path='/contact' element={<Contact />} />
      <Route path='/posts' element={<Posts />} />
      <Route path='/login' element={<Login />} />
      <Route path='/' element={<About />} />
      <Route path='/about' element={<About />} />
      <Route path='/registerUser' element={<Login />} />
      <Route path='/post' element={<Post />} />
      <Route path='/uploadPhotos' element={<UploadPhotos />} />
      <Route path='/secHome' element={<SecHome />} />

      <Route path='/' element={<Protected />}>
        <Route path='/createPost' element={<Admin />} />
        <Route path='/demo' element={<Counters />} />
        <Route path='/guessWord' element={<GuessedWordWrapper />} />
        <Route path='/jotto' element={<JottoApp />} />
        <Route path='/demo/Counters' element={<Counters />} />
        <Route path='/demo/Gallery' element={<Gallery />} />
        <Route path='/demo/To rtretrtdos' element={<todos />} />
        <Route path='/demo/Article' element={<Article />} />
        <Route path='/demo/Support' element={<Support />} />
        <Route path='/demo/counter' element={<Counters />} />
        <Route path='/users' element={<Users />} />
        <Route path='/demo/galaxy' element={<Galaxy />} />
        <Route path='/profile' element={<Profile />} />
      </Route>

      <Route path='/signup' element={<PublicRoutes />}></Route>
      <Route path='/unAuthUser' element={<UnAuthUser />}></Route>
    </Routes>
  );
};
