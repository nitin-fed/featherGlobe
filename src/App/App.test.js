import React from "react";
import App from "./App";
import { shallow, mount } from "enzyme";

describe("App", () => {
  it("Render", () => {
    const wrapper = mount(<App />);
    console.log(wrapper.debug());

    expect(wrapper.find(App).exists()).toBeTruthy();
    // expect(wrapper.find(Logo).exists()).toBeTruthy();
    // expect(wrapper.find(Navigation).exists()).toBeTruthy();
  });
});
