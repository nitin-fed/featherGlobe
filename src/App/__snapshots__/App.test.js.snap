// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render Render 1`] = `
<div
  className="mainWrapper"
>
  <div
    className="logo"
  >
    <p />
    <img
      alt="FeatherGlobe"
      className="logo"
      src="../images/globe.gif"
    />
  </div>
  <div>
    <ul
      className="appNav show"
    >
      <li>
        <a
          aria-current={null}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/profile"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/photography"
          onClick={[Function]}
        >
          Photography
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/demo"
          onClick={[Function]}
        >
          Demo
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/guessWord"
          onClick={[Function]}
        >
          Guess Word
        </a>
      </li>
    </ul>
    <hr />
    <div
      className="hamburger"
      onClick={[Function]}
    >
      <button>
        X
      </button>
    </div>
    <div />
  </div>
  <footer
    className="footer"
  >
    www.featherglobe.com
  </footer>
</div>
`;
